src/Grid.cpp

```diff
@@ -0,0 +1,114 @@
+#include "Grid.hpp"
```

Comment:
I would prefer to stick to one file naming convention. I use snake case and if it's not a problem for you I would prefer you to do the same.

Reply:
As I can see, Grid.hpp exports class named Grid. Files which exports classes could be named with PascalCase, isn't it ?

Reply:
> Files which exports classes could be named with PascalCase

I have never heard of this. It's not a huge deal, but I would like to keep one naming convention.

According to these coding conventions you should always use snake case:
https://google.github.io/styleguide/cppguide.html#File_Names
https://www.geeksforgeeks.org/naming-convention-in-c/

Reply:
Maybe this convention is only used in TypeScript and others languages excluding C++, but I think it's more readable when classes are named with PascalCase and variables/functions are named with camelCase.
I suggest using this convention.

[](https://builtin.com/articles/pascal-case-vs-camel-case)

Reply:
> Maybe this convention is only used in TypeScript and others languages excluding C++, but I think it's more readable when classes are named with PascalCase and variables/functions are named with camelCase. I suggest using this convention.
> 
> https://builtin.com/articles/pascal-case-vs-camel-case

Yeah classes are pascal case in C++ too, the variables are camel (member variables) or snake case (local variables) and the functions are pascal case (or snake case for small, general utility functions). Thats how I like to do this at least.